documentation/smg/17_using_pixel_data.md

```diff
@@ -0,0 +1,394 @@
+
+```matlab
+>> pix_signal = sqw_obj.data.pix(8, :);
+>> pix_coordinates = sqw_obj.data.pix(1:4, :);
+```
+
+This meant that accessing data required previous knowledge of the data structure;
+to get the signal, one needed to know it is stored in column 8.
+It also meant that actions performed on pixel data relied on the array's structure.
+This would cause issues if it was ever advantageous for the structure of the
+pixel data to be changed.
+
+#### Data API - Solution
+
+It was proposed that the `PixelData` class would provide getters/setters for
+pixel data fields.
+This would not assume users of the class had knowledge of the underlying data
+structure.
+It would also mean that, if the pixel data structure were to change,
+other code using pixel data could remain the same.
+
+### Extendability
+
+There was also a requirement that the pixel data be extendable,
+i.e. users should be able to add extra named columns to the data.
+
+#### Extendability - Solution
+
+The proposed `PixelData` class would allow addition of extra data per pixel,
+in named elements.
+These data could be retrieved using a getter that takes a char/string argument.
+
+### Parallelism
+
+As pixel data is often large, it's advantageous to perform operations on the
+data using distributed memory parallelism.
+To aid in this, a simple way of "chunking" the data between workers was required.
+
+#### Parallelism - Solution
+
+The ability to move to specific pages of pixel data was proposed to solve this
+problem.
+Using the `move_to_page(page_num)` method allows a worker to position itself at
+a particular point in the pixel data.
+This reduces distributing pixels between workers to a function of:
+the number of workers, the number of pixel, the page size.
+Each worker can receive its start and end page index from the head node,
+and operate on those pixels.
+
+## The PixelData Class
+
+### Principles
+
+- It operates with "pages" of cached data.
+  - The page size is specified on construction (in bytes).
+  - Only a page size worth of pixels is loaded at any one time.
+  - If all pixels fit in one page, all pixels are loaded.
+
+- Temporary files hold changes to pixel data.
+  - When pixel data is changed, the changes are written to a temporary file
+    when the current page is cleared.
+    This most often occurs when the next page is loaded.
+  - Changes are _never_ written to the SQW file used in construction.
+  - A logical array is held on the object that tracks which pages are "dirty".
+  - If a page is "clean" the data is loaded from the SQW file, if the page is
+    "dirty", it is loaded from a temporary file.
+  - Each page has its own temporary file.
+
+- It is a
+  [`handle` class](https://www.mathworks.com/help/matlab/matlab_oop/handle-objects.html).
+  - To copy a `PixelData` object, you must explicitly call the `copy` method.
+    This copy can be expensive, as all temporary files are also copied.
```

Comment:
Are they kept as temporaries or are they made part of the core? Does it still refer to the original SQW?